roles/forklift/tasks/main.yml

```diff
@@ -9,16 +9,34 @@
 
 - name: 'Check Forklift state'
   fail:
-    msg: 'The forklift_state variable is undefined. Please use pass "-e "forklift_state=up"" to spin up the boxes or "-e "forklift_state=destroy"" to destroy the boxes'
-  when: forklift_state is undefined
+    msg: 'The invalid value for forklift_state variable. Please use pass "-e "forklift_state=up"" to spin up the boxes or "-e "forklift_state=destroy"" to destroy the boxes'
+  when: forklift_state not in ('up', 'destroy')
 
-- name: 'Bring boxes up'
-  include_tasks: 'up.yml'
-  when: forklift_state == 'up'
+- name: 'Write box file'
+  copy:
+    dest: "{{ current_directory }}/boxes.d/80-tmp-{{ forklift_name }}.yaml"
+    content: "{{ forklift_boxes | to_yaml }}"
+  when: forklift_state == "up"
+
+- name: 'Bring up boxes'
+  command: "vagrant up {{ item.key }}"
+  args:
+    chdir: "{{ current_directory }}"
+  with_dict: "{{ forklift_boxes }}"
+  when: forklift_state == "up"
 
 - name: 'Destroy boxes'
-  include_tasks: 'destroy.yml'
-  when: forklift_state == 'destroy'
+  command: "vagrant destroy {{ item.key }}"
```

Comment:
I wonder if for this and `up`, we should instead do a single `vagrant destroy` with each box space separated? That might be more efficient than calling `vagrant` multiple times.

Reply:
Correct - I implemented that in f25188a9a258448a68a29b56c3662faf1c64625d and then I did a bad rebase on that.